src/services/consent/services/consentVersionService.js

```diff
@@ -64,17 +63,26 @@ class ConsentVersionService {
 		return this.app.service('consentVersionsModel').get(id, prepareInternalParams(params));
 	}
 
-	find(params) {
+	async find(params) {
 		const { query } = params;
-		if (query && query.schoolId) {
-			if (!query.$or) {
-				query.$or = [];
-			}
-			query.$or.push({ schoolId: query.schoolId });
-			delete query.schoolId;
+		const {
+			$limit, $sort, $skip, consentTypes,
+		} = query;
+
+		const baseQuery = {
+			$limit, $sort, $skip, consentTypes,
+		};
+		let searchResult;
+		if (query.schoolId) {
+			const queryWithSchool = { query: {...baseQuery, schoolId: query.schoolId }};
+			searchResult = await this.app.service('consentVersionsModel').find(prepareInternalParams(queryWithSchool));
 		}
 
-		return this.app.service('consentVersionsModel').find(prepareInternalParams(params));
+		if (searchResult.total > 0) {
+			return searchResult;
+		}
+		const querySchoolIdEmpty = { query: { ...baseQuery, schoolId: { $exists: false } }};
```

Comment:
Codacy found an issue: [A space is required before '}'.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5719243)